src/test/java/io/vertx/redis/client/test/RedisClusterTest.java

```diff
@@ -1163,6 +1164,70 @@ public void setAndWaitEmptyBatch(TestContext should) {
     ));
   }
 
+  @Test(timeout = 30_000)
+  @SuppressWarnings("unchecked")
+  public void batchSameSlotGroupByMultipleSlotsCommands(TestContext should) {
+    final Async test = should.async();
+
+    Map<Integer, List<Request>> mapCommands = new HashMap<>();
+    for (int i = 0; i < 100000; i++) {
+      String key = "key-" + i;
+      String value = "value-" + i;
+      int endpoint;
+      int slot = ZModem.generate(key);
+      if (slot < 16384 / (options.getEndpoints().size() / 2)) {
+        endpoint = 0;
+      } else if (slot <= 2 * 16384 / (options.getEndpoints().size() / 2)) {
+        endpoint = 1;
+      } else {
+        endpoint = 2;
+      }
```

Comment:
This is likely to work, but seems fragile. Wouldn't it be best to connect to the actual cluster, read `CLUSTER SLOTS` (likely using the `Slots` class) and merge the commands based on the actual current assignment?

Reply:
As main reason of using batch is for performance, waiting to execute a command just to perform batch seems counter intuitive. This should be done "offline" but I do agree this is fragile. I propose that we expose an additional helper function which would return endpoint of belonging key. This way we can use already known state of the cluster via Vert.x Cluster Redis APi

Reply:
Not sure I follow. What I propose is to read `CLUSTER SLOTS` _once_, before you start to group commands into batches, so that the grouping is based on actual reality. (Although the actual reality in a newly formed cluster is likely to follow the static assignment in this PR.)

Exposing this as an API is an interesting idea, but it would expand the scope of this PR significantly.

Reply:
I just wanted to have a viable code in test that shows how to perform grouping via hashslot without first contacting redis to obtain slot distribution as this will introduce a delay. As this is not documented I wanted to have it in tests as most of the times that is where I look on how to use some functionality. As long as you know the number of masters this is viable for grouping. When batch is submitted Vert.x driver will lookup hash lost accurately and decide on the endpoint. But ok, to keep it clean I have submitted the modified tests.